public/app/plugins/datasource/influxdb/components/queryUtils.test.ts

```diff
@@ -0,0 +1,287 @@
+            },
+          ],
+          hide: false,
+          measurement: 'cpu',
+          orderByTime: 'DESC',
+          policy: 'default',
+          rawQuery: false,
+          refId: 'A',
+          resultFormat: 'time_series',
+          select: [
+            [
+              {
+                type: 'field',
+                params: ['usage_idle'],
+              },
+              {
+                type: 'mean',
+                params: [],
+              },
+              {
+                type: 'holt_winters_with_fit',
+                params: ['30', '5'],
+              },
+            ],
+            [
+              {
+                type: 'field',
+                params: ['usage_guest'],
+              },
+              {
+                type: 'median',
+                params: [],
+              },
+            ],
+          ],
+          tags: [
+            {
+              key: 'cpu',
+              operator: '=',
+              value: 'cpu2',
+            },
+            {
+              condition: 'OR',
+              key: 'cpu',
+              operator: '=',
+              value: 'cpu3',
+            },
+            {
+              condition: 'AND',
+              key: 'cpu',
+              operator: '=',
+              value: 'cpu1',
+            },
+          ],
+          limit: '12',
+          slimit: '23',
+          tz: 'UTC',
+        })
+      ).toBe(
+        `SELECT holt_winters_with_fit(mean("usage_idle"), 30, 5), median("usage_guest") FROM "cpu" WHERE ("cpu" = 'cpu2' OR "cpu" = 'cpu3' AND "cpu" = 'cpu1') AND $timeFilter GROUP BY time($__interval), "cpu", "host" fill(none) ORDER BY time DESC LIMIT 12 SLIMIT 23 tz('UTC')`
+      );
+    });
+  });
+
+  describe('normalizeQuery', () => {
+    it('should handle minimal query', () => {
+      expect(
+        normalizeQuery({
+          refId: 'A',
+        })
+      ).toStrictEqual({
+        refId: 'A',
+        policy: 'default',
+        resultFormat: 'time_series',
+        orderByTime: 'ASC',
+        tags: [],
+        groupBy: [
+          { type: 'time', params: ['$__interval'] },
+          { type: 'fill', params: ['null'] },
+        ],
+        select: [
+          [
+            { type: 'field', params: ['value'] },
+            { type: 'mean', params: [] },
+          ],
+        ],
+      });
+    });
+
+    it('should not change values if they already exist', () => {
+      const query: InfluxQuery = {
+        refId: 'A',
+        groupBy: [],
+        measurement: 'cpu',
+        orderByTime: 'ASC',
+        policy: 'default',
+        resultFormat: 'table',
+        select: [
+          [
+            {
+              type: 'field',
+              params: ['usage_idle'],
+            },
+          ],
+        ],
+        tags: [],
+      };
+
+      // i want to make sure the code does not mutate the query
+      // by any chance, so we will use a deep-clone of the query
+      const cloneQuery = cloneDeep(query);
```

Comment:
I'm not sure i understand this, how are you checking the original object is not being mutated?

Reply:
i should rewrite that comment, what i meant is that i am making a clone of the query, and only giving the clone to the query-model-function, because it mutates and i do not want the original query to get mutated.

Reply:
oh, got it. 👍 makes sense!

Reply:
ignore my previous comment, i thought you are talking about a different file, sorry 😸 

i changed both the test-code and the comments there, it should be better now.